loris/ithaka.py

```diff
@@ -0,0 +1,82 @@
+import base64
+import gzip
+import itertools
+import json
+import os
+import random
+from StringIO import StringIO
+
+import requests
+from requests import HTTPError
+
+
+def get_session_id(session_text):
+    out = base64.urlsafe_b64decode(session_text+"===")
+    data = json.loads(gzip.GzipFile(fileobj=StringIO(out)).read())
+    return data['id']
+
+
+class DiscoverableService:
+
+    @staticmethod
+    def resolve(service_name):
+        eureka_url = "https://eureka.apps.%s.cirrostratus.org" % os.getenv("SGK_ENVIRONMENT", "test")
```

Comment:
I think you can avoid a hit to apps-gateway by foregoing part of the domain here:

```suggestion
        eureka_url = "https://eureka.%s.cirrostratus.org" % os.getenv("SGK_ENVIRONMENT", "test")
```

Reply:
I take it this app doesn't have a Eureka sidecar of its own?

Reply:
Doesn't seem to.

Reply:
If you expect service discovery to happen frequently, you may want to talk with CORE about the intricacies of having a sidecar vs. calling Eureka directly...I don't rightly know much in that arena, though my guess is that making `localhost` calls would be beneficial!